src/proto/h1/role.rs

```diff
@@ -271,7 +271,10 @@ impl Http1Transaction for Server {
                     }
                 }
                 header::EXPECT => {
-                    expect_continue = value.as_bytes() == b"100-continue";
+                    // According to https://datatracker.ietf.org/doc/html/rfc2616#section-14.20
+                    // Comparison of expectation values is case-insensitive for unquoted tokens
+                    // (including the 100-continue token)
+                    expect_continue = value.to_str().map(|s| s.eq_ignore_ascii_case("100-continue")).unwrap_or(false);
```

Comment:
`eq_ignore_ascii_case` should also available with `.as_bytes()`
See: https://doc.rust-lang.org/nightly/std/primitive.slice.html#method.eq_ignore_ascii_case

Reply:
Ah, that's great! Using that instead now.